tests/by-util/test_pgrep.rs

```diff
@@ -369,6 +369,34 @@ fn test_invalid_signal() {
         .stderr_contains("Unknown signal 'foo'");
 }
 
+#[test]
+#[cfg(target_os = "linux")]
+fn test_signals_that_never_match() {
+    new_ucmd!()
+        .arg("--require-handler")
+        .arg("--signal=0")
+        .arg(".*")
+        .fails()
+        .no_output();
```

Comment:
I think the assumption that the `0` signal never matches is incorrect. If I run the command with the original `pgrep`, and I have `top` running, it returns the PID of `top`.